Implement rule scatter plot function

Zapiano opened the pull request:
This plot may be useful for bump hunting in the future. It is a first  way of visualizing the rules extracted using SIRUS algorithm.

ConnectedSystems commented on ext/AvizExt/viz/rule_extraction.jl:
```diff
@@ -0,0 +1,114 @@
+            condition = rules[index].condition
+
+            # Feature names
+            condition_features = first.(condition)
+            foi = ms.fieldname .∈ [condition_features]
+            h_names = ms[foi, :name]
```
I think this extracts the factors/features of interest but does not check for order so it is possible for the figure axis labels to be in the wrong order

Zapiano commented:
@ConnectedSystems
I've added a legend, did some small refactoring and changed the color to something less aggressive. You can review it again now. 

![image](https://github.com/open-AIMS/ADRIA.jl/assets/8040719/300f2e5f-c075-4637-b0dc-36e00eb405f4)

ConnectedSystems commented:
I still argue the vertical/horizontal guide lines should be given more transparency.
Same with the hypercube/box/rectangle really.

ConnectedSystems commented on ext/AvizExt/viz/viz.jl:
```diff
@@ -43,8 +43,34 @@ function timesteps(outcomes::NamedDimsArray)::Array{Int64}
+# Returns
+Number of rows and columns
+"""
+function _calc_gridsize(n_factors::Int64; max_cols::Int64=4)::Tuple{Int64,Int64}
+    if n_factors <= 4
+        n_cols::Int64 = 2
+    else
+        n_cols = max_cols
+    end
+
+    n_rows::Int64 = ceil(Int64, n_factors / n_cols)
+
+    return n_rows, n_cols
+end
```
This needs to include the changes I raised earlier:

https://github.com/open-AIMS/ADRIA.jl/pull/425/files

Zapiano replied:
@ConnectedSystems Sorry about that, I fixed it now.

Zapiano commented:
> Looking good Pedro
> 
> I still argue the vertical/horizontal guide lines should be given more transparency. Same with the hypercube/box/rectangle really.

About that:

- I have added a transparency to the guide lines and made the target_area even more transparent.
- I have also made it optional to plot this _highlight_target_area. Below are two examples with this option equals true and false:

![rules_all_4096_6_true](https://github.com/open-AIMS/ADRIA.jl/assets/8040719/88e457d5-42f6-49df-aa21-fbe0480a5e30)

![rules_all_4096_6_false](https://github.com/open-AIMS/ADRIA.jl/assets/8040719/10dfa359-a5c4-4807-847a-d109beda7af0)

ConnectedSystems reviewed (approved):
Currently, this approach uses `target_cluster()` to determine which cluster to target. 

In future we will need something that accepts a pre-determined target cluster (or set of scenarios).
